bravado/mapping/param.py

```diff
@@ -128,6 +132,73 @@ def marshal_param(param, value, request):
             format(location))
 
 
+def unmarshal_param(param, request):
+    """Unmarshal the given parameter from the passed in request like object.
+
+    :type param: :class:`bravado.mapping.param.Param`
+    :type request: :class:`bravado.mapping.request.RequestLike`
+    """
+    param_spec = get_param_type_spec(param)
+    location = param.location
+    cast_param = partial(cast_request_param, param_spec['type'], param.name)
+
+    if location == 'path':
+        raw_value = cast_param(request.path.get(param.name, None))
```

Comment:
Yes, this is server side - `unmarshal` implies "wire -> python" and `param` implies request object. The only place that a request parameter is unmarshaled is on the server side.

To address your original question, a 3rd party server side request object doesn't usually know the types of the values that are extracted from the request path, query, form data, or headers. Hence, the type is implicitly a string. The casting is necessary to coerce the value into a type that matches the swagger spec.  If the casting fails, that is totally ok...the validator will catch it downstream.